add release notes (#60)
bisdn/bisdn-linux: release-notes.txt

BISDN Linux
4.7.0 Release Notes
Date: 2022-08-03

! New in this release
- Yocto: Updated Yocto from 3.1 (dunfell) to 4.0 (kirkstone).
  This includes major version updates for most OS components.
- Linux kernel: Updated to 5.15.56.
- FRR: Updated to 8.3.

! Changed behavior

> - MAC addresses of port* interfaces are now set based on EEPROM contents instead

Reviewer: ```suggestion
- MAC addresses of Port* interfaces are now set based on EEPROM contents instead
```

Author: The interfaces use lower case names, so I think this should stay lower case.

  of randomly generated.

> - The default OF-DPA L2L3 allocation mode is now balanced, increasing the amount

Reviewer: ```suggestion
- The default OF-DPA L2/L3 allocation mode is now balanced, increasing the amount
```

Author: Fixed.

  of possible L2 and L3 entries on Trident 3 based platforms.

! Bug fixes

> - mstpd: Fixed an issue causing missed netlink events like link change changes

Reviewer: ```suggestion
- mstpd: Fixed an issue causing missed netlink events like link state changes
```

Author: Fixed.

  of ports, resulting in a desync between mstpd state and actual/kernel state.

> - The installer will now rename default sudoers configuration files in way that

Reviewer: ```suggestion
- The installer will now rename default sudoers configuration files in a way that
```

Author: Added.

update layer to kirkstone
onl: ship onlpd instead of onlpdump
kernel: update source comments in linux-yocto-onl_5.15.bb
conf: drop BISDN_ONIE_PLATFORM now that it is autogenerated
onl: move ONL_DEBIAN_SUITE into recipe
conf: drop now unused ONL_ARCH
onl: calculate ONL_BUILD_ARCH dynamically
conf/machine: drop ONL_PLATFORM_SUBDIR
onl.inc: drop unused PLATFORM_PATH
onl: use the onl package manager for resolving paths
onl: update python3 conversion patch
conf: fixup override usage
linux-yocto-onl: do not set CONFIG_IP_NF_FILTER
linux-yocto-onl: do not set CONFIG_NET_UDP_TUNNEL
linux-yocto-onl: x86-intel: do not set CONFIG_CGROUP_HUGETLB
linux-yocto-onl: x86-intel: do not set CONFIG_VXLAN
linux-yocto-onl: use kernel-meta for ext4
linux-yocto-onl: armel-iproc: use ext4 module for ext2 and ext3
linux-yocto-onl: armel-iproc: enable FS_SECURITY for ext4
linux-yocto-onl: armel-iproc: enable ACL support ext4
onl: make use of yocto's python3-native
yocto-linux-onl: armel-iproc: do not disable VXLAN

meta-switch:
Revert "frr: update to 8.3"
systemd: let any interface satisfy systemd-networkd-wait-online
bump distro to version 3.7.0
baseboxd: bump to 1.12.0

> frr: update to 8.3

Reviewer 2: I would prefer to just delete both commit messages regarding 8.3 (commit and revert), since otherwise people might only see the first one and be a bit confused.

Author: I'd rather not "hand-tune" autogenerated things. You will also see it when you look at the git history yourself, so I think having it is fine.

Reviewer 2: I'd rather not cofuse people that read this and do not know that this is auto generated.

Reviewer 2: Or is this not not exatly the same that we will upload to the repos?

Author: How about we change the title to "Detailed changelog (auto generated from git history)" or so? This really should be a fire and forget thing, and not be touched up by anything. Any manual changes will also get likely be dropped/missed in case we need to regenerate it.

Reviewer 2: We changed this a couple of times manually before, to avoid confusing people or to redact certain internal information. Lets just remove the FRR 8.3 here, so it does not cause any confusion for people grepping through the changelog.

Author: I still disagree, but changed it to not getting hung up on this.

BISDN Linux
4.7.0 Release Notes

> Date: 2022-08-03

Reviewer: Don't forget to update this.

Author: Updated to friday.

! New in this release
- Yocto: Updated Yocto from 3.1 (dunfell) to 4.0 (kirkstone).
  This includes major version updates for most OS components. For a detailed
  list of changes, see [1] and the release notes for 3.2 to 3.4.

- Linux kernel: Updated to 5.15.56.

> - FRR: Updated to 8.3.

Reviewer: This needs an update.

Author: Dropped it, and added the revert commit to meta-switch.
